wlauto/workloads/skypeecho/__init__.py

```diff
@@ -0,0 +1,143 @@
+#    Copyright 2014-2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os.path as op
+import time
+
+from wlauto import AndroidUiAutoBenchmark, Parameter
+
+
+SKYPE_ACTION_URIS = {
+    'call': 'call',
+    'video': 'call&video=true',
+}
+
+
+class SkypeEcho(AndroidUiAutoBenchmark):
+
+    name = 'skypeecho'
+    description = '''
+    Workload that makes Skype calls
+
+    It allows for the agenda to decide whether to make a voice call or a video call.
+    Credentials for the user account used to log into the Skype app have to be provided
+    in the agenda, as well as the display name and skype ID of the contact to call.
+
+    Other optional arguments allow controlling the duration of the call, whether the
+    call includes video or voice only, and whether to collect sys dumps.
+
+    For reliable testing, this workload requires a good and stable internet connection,
+    preferably on Wi-Fi.
+    '''
+    package = 'com.skype.raider'
+    activity = ''
+    # Skype has no default 'main' activity
+    launch_main = False # overrides extended class
```

Comment:
I'm a bit confused by the requirement for this.  The Skype test has use_gui set true so how do you get to the point where you require the use of specific intents?